### PR TITLE
Added stub for siginterrupt

### PR DESCRIPTION
**Pull request by sappigeninja**
I added a stub, as sigaction is not yet implemented.

### CODE REVIEW

**Comment by Dennisbonke** [comment]
sigaction is implemented, and can be found here: https://github.com/managarm/mlibc/blob/ea4a158a3cd9608fdcd4b981f05b29739e0298de/options/posix/generic/posix_signal.cpp#L64-L75

**Comment by sappigeninja** [comment]
> sigaction is implemented, and can be found here:
> 
> https://github.com/managarm/mlibc/blob/ea4a158a3cd9608fdcd4b981f05b29739e0298de/options/posix/generic/posix_signal.cpp#L64-L75

I used this to implement it:
https://www.systutorials.com/docs/linux/man/3p-siginterrupt/

File: options/posix/generic/posix_signal.cpp [hunk 1]
```diff
@@ -74,6 +74,20 @@ int sigaction(int signum, const struct sigaction *__restrict act, struct sigacti
...
 
+int siginterrupt(int sig, int flag) {
+    int ret;
+    struct sigaction act;
+
+    sigaction(sig, NULL, &act);
+    if (flag)
+        act.sa_flags &= ~SA_RESTART;
+    else
+        act.sa_flags |= SA_RESTART;
```
Comment (avdgrinten) [inline options/posix/generic/posix_signal.cpp:85]: Don't we need to clear the bit in any case?

Comment (sappigeninja) [reply options/posix/generic/posix_signal.cpp:85]: I dont understand, could you clarify?

Comment (avdgrinten) [reply options/posix/generic/posix_signal.cpp:85]: Oh, nevermind, I just had a mental lapse.

**Comment by avdgrinten** [comment]
LGTM, thank you for the PR!

**Pull request merged**